### PR TITLE
CI: check for changes to autogenerated files

### PR DESCRIPTION
**Pull request by EricFromCanada**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----
A basic CI job that checks if changes to autogenerated files don't reflect their source files, as [suggested here](https://github.com/Homebrew/brew/pull/14714#pullrequestreview-1305439063).

### CODE REVIEW

File: .github/workflows/autogenerated-files.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      - name: Check for changes to autogenerated files
+        id: check
+        run: |
+          if brew generate-man-completions
+          then
+            echo "This PR modifies autogenerated files!" >&2
+            echo "Please ensure their source files are updated and then run the following:
+              brew generate-man-completions
+              brew update-maintainers
+              brew update-sponsors" >&2
```
Comment (Bo98) [inline .github/workflows/autogenerated-files.yml:44]: I think most users can't run the last command.

**Review by Rylan12 — APPROVED** [review]
I think this makes sense! We can adjust if we end up finding this disruptive

**Comment by MikeMcQuaid** [comment]
Great work here @EricFromCanada! It occurred to me this is very similar to what `vendored-gems` is doing so I added it to that job rather than add a new one.

**Comment by MikeMcQuaid** [comment]
> It occurred to me this is very similar to what vendored-gems is doing so I added it to that job rather than add a new one.

This was a bad idea 😅. Because of how many repos can break this, it's likely to break a required job pretty often. Reverting!

**Pull request merged**